src/natpmp.cpp

```diff
@@ -315,12 +314,11 @@ void natpmp::update_mapping(int const i)
 	if (should_log())
 	{
 		log("update-mapping: proto: %s port: %d local-port: %d action: %s ttl: %" PRId64
-			, (m.protocol == portmap_protocol::none
-				? "none" : m.protocol == portmap_protocol::tcp ? "tcp" : "udp")
+			, m.protocol == portmap_protocol::none
+				? "none" : aux::to_string(m.protocol)
```

Comment:
I don't believe you need to specify ``aux::`` when calling ``to_string``. It should pick the right function by ADL